### PR TITLE
feat!: Add login query param support to ListCredentialAuthorizations

### PR DESCRIPTION
**Pull request by zeusdeux**
BREAKING CHANGE: `ListCredentialAuthorizations` now takes `opts *CredentialAuthorizationsListOptions` instead of `ListOptions`.

As per the docs for the [List SAML SSO authorizations for an organization](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/orgs?apiVersion=2022-11-28#list-saml-sso-authorizations-for-an-organization) endpoint — it supports a query param named `login` to filter the authorizations by a github user.

This PR adds support for passing this `login` query param via a new struct (`CredentialAuthorizationsListOptions`) which embeds `ListOptions` as `ListOptions` itself only supports `Page` and `PerPage` query params.

### CODE REVIEW

File: github/orgs_credential_authorizations.go
Comment (zeusdeux) [inline github/orgs_credential_authorizations.go:63]: Chose to embed `ListOptions` instead of extending it as it is reused for other `List*` functions which don't necessarily support the `login` query parameter.

File: github/orgs_credential_authorizations.go
Comment (gmlewis) [inline github/orgs_credential_authorizations.go:60]: nit:

```suggestion
// such as Page and PerPage.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @zeusdeux!
One tiny nit, please, for the generated `godocs`, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by zeusdeux** [comment]
Thanks for the quick review @gmlewis! Accepted your suggestion!

**Review by gmlewis — APPROVED** [review]
Thank you, @zeusdeux !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by zeusdeux** [comment]
@gmlewis should I tag someone as a second reviewer?

**Comment by gmlewis** [comment]
> @gmlewis should I tag someone as a second reviewer?

Since this repo is maintained strictly by volunteers, I try to not tag individuals to respect their time unless there is an urgent need.
Having said that, please feel free to perform code reviews on any PRs that have been labeled "NeedsReview".

**Comment by zeusdeux** [comment]
Sounds good @gmlewis! Thank you for the information. 
Btw, re: failing build — any notes on the codecov/project failure? Not sure if I should take that into account given the test change in this PR covers the new code path. 🙏🏽

**Comment by gmlewis** [comment]
> Btw, re: failing build — any notes on the codecov/project failure? Not sure if I should take that into account given the test change in this PR covers the new code path. 🙏🏽

Unfortunately, you can ignore the Codecov report, as we are waiting for #3221 to be addressed by @google-admin and update this repo's Codecov authentication token so that our reports will be useful.

**Comment by zeusdeux** [comment]
Makes sense. Does that mean that this PR only needs one more ✅ and it'll be good to land? 
And if so, what's the release process? Thanks for all the help so far @gmlewis! 🙏🏽

**Review by alexef — APPROVED** [review]
lgtm

**Comment by gmlewis** [comment]
> Does that mean that this PR only needs one more ✅ and it'll be good to land?

Yes, and it is now ready to merge. Thank you, @alexef !

> And if so, what's the release process? 

We cut a new release about once per month, so the next release should be in three to four weeks. 

Merging.

**Comment by zeusdeux** [comment]
Thank you @gmlewis and @alexef! 🙏🏽

**Pull request merged**
